### PR TITLE
Python: Add syntax example comments for automatic document generation.

### PR DESCRIPTION
**Pull request by markshannon**
Adds comments of the form `/* syntax: ... */` to all relevant AST classes, to improve automatically generated documentation .

### CODE REVIEW

File: python/ql/src/semmle/python/Stmts.qll [hunk 1]
```diff
@@ -260,6 +281,8 @@ class ExprStmt extends ExprStmt_ {
 /** A for statement, such as `for x in y: print(x)` */
 class For extends For_ {
 
+    /* syntax: for varname in Expr: */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Stmts.qll]: I'm surprised by the lack of `...` in this and the `async for` statement below.

File: python/ql/src/semmle/python/AstExtended.qll [hunk 2]
```diff
@@ -3,6 +3,9 @@ import python
 /** Syntactic node (Class, Function, Module, Expr, Stmt or Comprehension) corresponding to a flow node */
 abstract class AstNode extends AstNode_ {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/AstExtended.qll]: What purpose does this comment serve? It only appears in _some_ cases, and I don't immediately see what the pattern is.

File: python/ql/src/semmle/python/Stmts.qll (hunk 1 above)
Comment (markshannon) [reply python/ql/src/semmle/python/Stmts.qll]: done

File: python/ql/src/semmle/python/AstExtended.qll (hunk 2 above)
Comment (markshannon) [reply python/ql/src/semmle/python/AstExtended.qll]: This is the root of the heirarchy. So it should only be here. I've clarified this one and deleted the others.

File: python/ql/src/semmle/python/Exprs.qll [hunk 3]
```diff
@@ -131,6 +131,9 @@ class Expr extends Expr_, AstNode {
 /** An attribute expression, such as `value.attr` */
 class Attribute extends Attribute_ {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 4]
```diff
@@ -456,6 +481,9 @@ class FloatLiteral extends Num {
 class ImaginaryLiteral extends Num {
     private float value;
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 5]
```diff
@@ -513,6 +541,9 @@ class NegativeIntegerLiteral extends ImmutableLiteral, UnaryExpr {
    "hello" are treated as Bytes for Python2, but Unicode for Python3. */
 class Unicode extends StrConst {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 6]
```diff
@@ -541,6 +572,9 @@ class Unicode extends StrConst {
 /** A dictionary expression, such as `{'key':'value'}` */
 class Dict extends Dict_ {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 7]
```diff
@@ -566,6 +600,9 @@ class Dict extends Dict_ {
 /** A list expression, such as `[ 1, 3, 5, 7, 9 ]` */
 class List extends List_ {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 8]
```diff
@@ -575,6 +612,9 @@ class List extends List_ {
 /** A set expression such as `{ 1, 3, 5, 7, 9 }` */
 class Set extends Set_ {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 9]
```diff
@@ -705,6 +749,9 @@ class Slice extends Slice_ {
 /** A string constant. */
 class StrConst extends Str_, ImmutableLiteral {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 10]
```diff
@@ -790,6 +837,9 @@ abstract class BooleanLiteral extends NameConstant {
 /** The boolean named constant `True` */
 class True extends BooleanLiteral {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 11]
```diff
@@ -807,6 +857,9 @@ class True extends BooleanLiteral {
 /** The boolean named constant `False` */
 class False extends BooleanLiteral {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 12]
```diff
@@ -824,6 +877,9 @@ class False extends BooleanLiteral {
 /** `None` */
 class None extends NameConstant {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

File: python/ql/src/semmle/python/Exprs.qll [hunk 13]
```diff
@@ -840,6 +896,9 @@ class None extends NameConstant {
 /** An await expression such as `await coro`. */
 class Await extends Await_ {
 
+    /* Special comment for documentation generation */
```
Comment (taus-semmle) [inline python/ql/src/semmle/python/Exprs.qll]: Here.

**Pull request merged**